### PR TITLE
change the operator pods' checking

### PR DESCRIPTION
**Pull request by jianzhangbjz**
**Description of the change:**
Adress some issues in the [merged PR 3819](https://github.com/operator-framework/operator-sdk/pull/3819)

**Motivation for the change:**

- `Succeeded` means the Pod finished and exited. For the operator pods, they should be running all the time to monitor the CRs, so, the expected status should be `Running`, not `Succeeded`.

- Sometimes, there are multi containers in a Pod. Should add the container name to distinguish them.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
@bharathi-tenneti @jmrodri @estroz Could you help review it? Thanks!

**Comment by bharathi-tenneti** [comment]
@jianzhangbjz Good catch, You can go ahead an remove the comments.

**Review by rashmigottipati — APPROVED** [review]

**Comment by jianzhangbjz** [comment]
Fix the below error:
```console
- /target/docs/contribution-guidelines/testing/travis-build/index.html
  *  External link https://github.com/operator-framework/operator-sdk/blob/master/test/e2e/e2e_suite_test.go failed: 404 No error
```

**Comment by jianzhangbjz** [comment]
@bharathi-tenneti Thanks! Removed the comments.

**Comment by jianzhangbjz** [comment]
@jmrodri @bharathi-tenneti Could you help add the `/lgtm` label to merge it? Thanks!

**Comment by jianzhangbjz** [comment]
@jmrodri @varshaprasad96 Could you help approve it? Thanks!

File: internal/olm/client/client.go [hunk 1]
```diff
@@ -280,10 +280,10 @@ func (c Client) printPodErrors(ctx context.Context, depSelectors *metav1.LabelSe
 		return fmt.Errorf("error getting Pods: %v", err)
 	}
 	for _, p := range podList.Items {
-		if p.Status.Phase != corev1.PodSucceeded {
+		if p.Status.Phase != corev1.PodRunning {
 			for _, cs := range p.Status.ContainerStatuses {
 				if !cs.Ready {
-					podErrors[p.Name] = cs.State.Waiting.Message
+					podErrors[p.Name+cs.Name] = cs.State.Waiting.Message
```
Comment (joelanford) [inline internal/olm/client/client.go:286]: Do we print this key later? If so, we should probably make it look nicer. E.g.
```suggestion
					podErrors[fmt.Sprintf("%s.%s", p.Name, cs.Name)] = cs.State.Waiting.Message
```"

Comment (jianzhangbjz) [reply internal/olm/client/client.go:286]: Hi @joelanford  yes, thanks!

**Comment by jianzhangbjz** [comment]
@joelanford Thanks for your suggestions! I added it. Could you help have a review again? Thanks!

**Comment by bharathi-tenneti** [comment]
@jianzhangbjz just FYI, this code is being refactored. https://github.com/operator-framework/operator-sdk/pull/3908
May be, you can review that PR, and see if it covers your changes.

**Review by camilamacedo86 — COMMENTED** [review]
the change shows fine. 
/lgtm

**Comment by jianzhangbjz** [comment]
Thanks! How to address this failure? 
```console
    TASK [Fetch the kubeconfig] ****************************************************
fatal: [kind-test-local]: FAILED! => {"changed": false, "file": "/root/.kube/config", "msg": "unable to calculate the checksum of the remote file"}
    
    PLAY RECAP *********************************************************************
    kind-test-local            : ok=0    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
    
ERROR: 
```
image

**Comment by jianzhangbjz** [comment]
@rafaelvanoni @bharathi-tenneti Could you help look at it? Thanks!

**Review by estroz — APPROVED** [review]
/lgtm

**Comment by estroz** [comment]
@jianzhangbjz can you rebase this with master? There was a CI change that isn't being picked up here.

File: internal/olm/client/client.go [hunk 2]
```diff
@@ -280,10 +280,10 @@ func (c Client) printPodErrors(ctx context.Context, depSelectors *metav1.LabelSe
 		return fmt.Errorf("error getting Pods: %v", err)
```
Comment (camilamacedo86) [inline internal/olm/client/client.go:280]: Could you please add the new changelog fragment in changelog/fragments (see changelog/fragments/00-template.yaml) for we are able to get it merged?

**Comment by camilamacedo86** [comment]
Hi @jianzhangbjz, 


Could you please rebase with the master and add the new changelog fragment in changelog/fragments (see changelog/fragments/00-template.yaml) for we are able to get it merged?

**Comment by jianzhangbjz** [comment]
Hi @camilamacedo86, I'm so sorry for the late reply. I forgot it to update it. The new logic looks good to me. But, I combine an `if` logic there, could you have a review? Thanks!

**Pull request closed without merging**